pkg/graveler/graveler.go

```diff
@@ -1286,6 +1293,26 @@ func (g *KVGraveler) SaveGarbageCollectionCommits(ctx context.Context, repositor
 	}, err
 }
 
+func (g *KVGraveler) SaveGarbageCollectionUncommitted(ctx context.Context, repository *RepositoryRecord, filename string, runID *string) (*GarbageCollectionRunMetadata, error) {
+	rID := g.garbageCollectionManager.NewID()
+	if runID != nil {
+		rID = *runID
+	}
```

Comment:
Make runID a string and then:
```suggestion
	if runID == "" {
		runID = *g.garbageCollectionManager.NewID()
	}
```

Reply:
Talked with @nopcoder agreed that this will be handled in the controller